src/jit/codegenxarch.cpp

```diff
@@ -3756,12 +3759,7 @@ void CodeGen::genCodeForCmpXchg(GenTreeCmpXchg* tree)
 // generate code for BoundsCheck nodes
 void CodeGen::genRangeCheck(GenTree* oper)
 {
-#ifdef FEATURE_SIMD
-    noway_assert(oper->OperGet() == GT_ARR_BOUNDS_CHECK || oper->OperGet() == GT_SIMD_CHK);
-#else  // !FEATURE_SIMD
-    noway_assert(oper->OperGet() == GT_ARR_BOUNDS_CHECK);
-#endif // !FEATURE_SIMD
-
+    assert(oper->OperIsBoundsCheck());
```

Comment:
Should this be kept as a `noway_assert`?

Reply:
Not sure, but will fix.

Reply:
No, `noway_assert` is pointless here. All other code gen functions use `assert`.